crates/miden-tx/src/prover/mod.rs

```diff
@@ -120,23 +120,22 @@ impl LocalTransactionProver {
         tx_witness: TransactionWitness,
     ) -> Result<ProvenTransaction, TransactionProverError> {
         let TransactionWitness {
-            tx_inputs,
-            tx_args,
+            mut tx_inputs,
             foreign_account_code,
             advice_witness,
         } = tx_witness;
 
-        let (stack_inputs, advice_inputs) =
-            TransactionKernel::prepare_inputs(&tx_inputs, &tx_args, Some(advice_witness))
-                .map_err(TransactionProverError::ConflictingAdviceMapEntry)?;
+        tx_inputs.extend_advice_inputs(advice_witness);
+        let (stack_inputs, advice_inputs) = TransactionKernel::prepare_inputs(&tx_inputs)
+            .map_err(TransactionProverError::ConflictingAdviceMapEntry)?;
```

Comment:
It's a bit awkward to carry the tx inputs and advice witness here separately _and then_ combine them. But also, as @bobbinth mentioned, this will likely be refactored anyway, so no big deal.